packages/dev/gui/src/2D/controls/stackPanel.ts

```diff
@@ -165,10 +163,8 @@ export class StackPanel extends Container {
                     child._top.ignoreAdaptiveScaling = true;
                 }
 
-                if (child._height.isPercentage && !child._automaticSize && !(child as TextBlock).resizeToFit && !(child as Container).adaptHeightToChildren) {
-                    if (!this.ignoreLayoutWarnings) {
-                        Tools.Warn(`Control (Name:${child.name}, UniqueId:${child.uniqueId}) is using height in percentage mode inside a vertical StackPanel`);
-                    }
+                if (!this.isHeightFullyDefined() && !this.ignoreLayoutWarnings) {
+                    Tools.Warn(`Control (Name:${child.name}, UniqueId:${child.uniqueId}) is using height in percentage mode inside a vertical StackPanel`);

```

Comment:
Isn't there a chance that this will execute on each frame? Can we trigger it only once in this case? or is it not possible?

Reply:
It does execute on each frame, I cache the width/height values but I can also add a check here to display the warning only once, which might be good for the user too.

Reply:
If you want, you can use `Logger.Warn(message, limit)` and set `limit` to the max number of times a message will be generated in the console (after the last occurrence, a message like `Too many %TYPE%s (%LIMIT%), no more %TYPE%s will be reported for this message.` will be generated so that the user knows we reached the limit for this message).

Reply:
No, I think I'm the only one to use this feature in the WebGPU engine. So, you are free to use the limit you want :)

Reply:
We can add the optional limit to Tools.Warn/Error/Log as well, as it is just a proxy to the logger. But you can use the logger directly without changing tools.

Reply:
We should limit our use of tool directly when not necessary anyway for tree shaking purpose.